vault/plan.sh

```diff
@@ -0,0 +1,30 @@
+pkg_origin=core
+pkg_name=vault
+pkg_version=0.6.2
+pkg_maintainer='The Habitat Maintainers <humans@habitat.sh>'
+pkg_license=("Apache 2")
+pkg_upstream=https://www.vaultproject.io/
+pkg_source=https://releases.hashicorp.com/vault/${pkg_version}/vault_${pkg_version}_linux_amd64.zip
+pkg_shasum=91432c812b1264306f8d1ecf7dd237c3d7a8b2b6aebf4f887e487c4e7f69338c
+pkg_filename=${pkg_name}-${pkg_version}_linux_amd64.zip
+pkg_deps=()
+pkg_build_deps=(core/unzip)
+pkg_bin_dirs=(bin)
+pkg_include_dirs=(include)
+pkg_lib_dirs=(lib)
+pkg_expose=(8200)
+pkg_svc_user=root
+
+do_unpack() {
+  cd "${HAB_CACHE_SRC_PATH}" || exit
+  unzip ${pkg_filename} -d "${pkg_name}-${pkg_version}"
+  rm ${pkg_filename}
```

Comment:
In earlier iterations, the zip file was getting copied to `$pkg_prefix`. That's the directory that is collecting the files installed for packaging. So then, yes, the zip would need to be deleted to not have it show up in the `.hart`.

With the latest set of changes and`do_download` no longer overridden:

* the zip file lands in `${HAB_CACHE_SRC_PATH}` (`/hab/cache/src`)
* unzipped into `${HAB_CACHE_SRC_PATH}/${pkg_name}-${pkg_version}` (`/hab/cache/src/vault-0.6.2`)
* the `install` command in `do_install` copies the `vault` binary to `"${pkg_prefix}"/bin/vault` (`/hab/pkgs/<origin>/vault/0.6.2/<timestamp>/bin/vault`)
* `/hab/pkgs/<origin>/vault/0.6.2/<timestamp>` directory gets packaged up into the `.hart` being built